api/data_explorer/util/elasticsearch_util.py

```diff
@@ -215,13 +240,20 @@ def get_elasticsearch_facet(es, elasticsearch_field_name, field_type):
             field=elasticsearch_field_name,
             interval=_get_bucket_interval(field_range))
 
-    # Handle sample facets in a special way since they are nested objects.
-    if elasticsearch_field_name.startswith('samples.'):
-        es_facet = NestedFacet('samples', es_facet)
+    nested_path = is_nested_facet(elasticsearch_field_name, nested_facet_paths)
```

Comment:
This will need to be done recursively. If you have `fieldA` which is RECORD type and that has inner `fieldB` which is also RECORD type, that will need to be translated into: 
```
NestedFacet('fieldA', NestedFacet('fieldB', es_facet))
```

Reply:
Malathi, can you please set up a Data Explorer to manually test this? Either use an existing table or create one.  Please let me know the table so I can test too. Thanks.

Also, I thought you had to traverse nested_facet_paths backwards, for this case.